src/kontrol/prove.py

```diff
@@ -337,6 +344,14 @@ def create_kcfg_explore() -> KCFGExplore:
                 )
 
             if proof is None:
+                # With CSE, top-level proof should be a summary if it's not a test or setUp function
+                if (
+                    options.cse
+                    and options.config_type == ConfigType.TEST_CONFIG
+                    and not test.method.contract_name.endswith('Test')
```

Comment:
This might not hold for all test contracts. I don't think it is enforced in any way.

Reply:
My suggestion would be to create a static function in the `Contract` class, that checks the AST of the contract to see if a contract inherits the `Test` contract, or if it has the `IS_TEST` variable in the storage. https://github.com/foundry-rs/forge-std/blob/master/src/Test.sol#L32

Reply:
Thanks @anvacaru! That's my concern too, but we need a way to determine if a constructor belongs to a test contract, and I couldn't think of a better one. It seems to be the case with CSE tests, because in https://github.com/runtimeverification/kontrol/pull/563 Noah used it as a way to decide on the configuration to be used for each test in `test_kontrol_cse`, and it's passing: https://github.com/runtimeverification/kontrol/blob/0b1e62950172f9101b589dd029374859456e38c2/src/tests/integration/test_kontrol_cse.py#L42-L43
But it's also getting stuck somewhere now, so I'm trying to isolate what it is.

Reply:
Oh wait, sorry, I guess it doesn't? https://github.com/runtimeverification/kontrol/actions/runs/9384264356/job/25839912502?pr=599

Reply:
> My suggestion would be to create a static function in the Contract class, that checks the AST of the contract to see if a contract inherits the Test contract, or if it has the IS_TEST variable in the storage. https://github.com/foundry-rs/forge-std/blob/master/src/Test.sol#L32

Yes, agreed, that sounds great! Will do it today.

Reply:
@anvacaru done in https://github.com/runtimeverification/kontrol/pull/599/commits/6351549a4c358edcc3b7627c311e5a7ea8b3a33e. Thanks for the suggestion!

I've decided to go with `IS_TEST` and leverage the `fields` you've added for it, since some Kontrol test contracts e.g., for [Optimism](https://github.com/ethereum-optimism/optimism/blob/c38ce096def52e4acdaecb7ffd2d396f464693dd/packages/contracts-bedrock/test/kontrol/proofs/OptimismPortal.k.sol#L24), don't inherit `Test` and it's easier to add an `IS_TEST` variable to them.

Reply:
And all tests are passing locally now.